File: codeforces/Equal--Subsequences.py
def solve(n, k):
    ones = k
    zeros = n - k
    s = []

    # Interleave 1s and 0s
    while ones > 0 and zeros > 0:
        s.append('1')
        ones -= 1
        if zeros > 0:
            s.append('0')
            zeros -= 1

    # Add remaining 1s or 0s
    s.extend(['1'] * ones)
    s.extend(['0'] * zeros)

    print(''.join(s))

# Read input
t = int(input())
for _ in range(t):
    n, k = map(int, input().split())
    solve(n, k)

File: codeforces/Equal_Subsequences.cpp
#include <iostream>
using namespace std;

void solve(int n, int k) {
    int ones = k;
    int zeros = n - k;
    string s = "";

    // Interleave 1s and 0s as long as we can
    while (ones > 0 && zeros > 0) {
        s += '1';
        ones--;
        if (zeros > 0) {
            s += '0';
            zeros--;
        }
    }

    // Add remaining 1s or 0s
    s += string(ones, '1');
    s += string(zeros, '0');

    cout << s << endl;
}

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n, k;
        cin >> n >> k;
        solve(n, k);
    }
    return 0;
}
